crates/nargo/src/cli/mod.rs

```diff
@@ -101,6 +101,40 @@ pub fn prove_and_verify(proof_name: &str, prg_dir: &Path, show_ssa: bool) -> boo
     }
 }
 
+// helper function which tests noir programs by trying to compile, generate a proof and verify it
+pub fn cached_prove_and_verify(
```

Comment:
I'm a little wary of ~2x-ing the length of CI time to check that we're saving and loading the cached keys correctly. We don't actually have to prove and verify here to test this. We then don't even need consistent set of ACIR + keys, i.e no need to compile and we can just use dummy ACIR + keys.

At this point we can then make this a unit rather than integration test.

Reply:
2x is a lot, you are right. I'm looking at how to make the test faster